### PR TITLE
GH-25: Add PhoneDialer API

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change ###

Added the API to open the phone dialer.

### Bugs Fixed ###

- Related to issue #25
- Old PR #37 

### API Changes ###

```csharp
static class PhoneDialer {
    static void Open(string number);
}
```

### Behavioral Changes ###

Added the PhoneDialer API.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Updated documentation ([see walkthrough](https://github.com/xamarin/Caboodle/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

File: Samples/Caboodle.Samples/View/PhoneDialerPage.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using Xamarin.Forms;
+using Xamarin.Forms.Xaml;
+
+namespace Caboodle.Samples.View
+{
+    [XamlCompilation(XamlCompilationOptions.Compile)]
```
Comment (mattleibow) [inline Samples/Caboodle.Samples/View/PhoneDialerPage.xaml.cs]: This is not needed.

File: Samples/Caboodle.Samples/ViewModel/HomeViewModel.cs [hunk 2]
```diff
@@ -11,12 +11,12 @@ public HomeViewModel()
             Items = new ObservableCollection<SampleItem>
             {
                 new SampleItem("Battery", typeof(BatteryPage), "Easily detect battery level, source, and state."),
-                new SampleItem("Browser", typeof(BrowserPage), "Quickly and easily open a browser to a specific website."),
```
Comment (mattleibow) [inline Samples/Caboodle.Samples/ViewModel/HomeViewModel.cs]: Why is this gone?

File: docs/en/index.xml [hunk 3]
```diff
@@ -3,13 +3,13 @@
...
         <Attribute>
-          <AttributeName>System.Diagnostics.Debuggable(System.Diagnostics.DebuggableAttribute+DebuggingModes.IgnoreSymbolStoreSequencePoints)</AttributeName>
+          <AttributeName>System.Diagnostics.Debuggable(System.Diagnostics.DebuggableAttribute+DebuggingModes.Default | System.Diagnostics.DebuggableAttribute+DebuggingModes.DisableOptimizations | System.Diagnostics.DebuggableAttribute+DebuggingModes.IgnoreSymbolStoreSequencePoints | System.Diagnostics.DebuggableAttribute+DebuggingModes.EnableEditAndContinue)</AttributeName>
         </Attribute>
         <Attribute>
           <AttributeName>System.Reflection.AssemblyCompany("Microsoft")</AttributeName>
         </Attribute>
         <Attribute>
-          <AttributeName>System.Reflection.AssemblyConfiguration("Release")</AttributeName>
+          <AttributeName>System.Reflection.AssemblyConfiguration("Debug")</AttributeName>
```
Comment (mattleibow) [inline docs/en/index.xml]: Docs should be generated on a release build - or just remove these two changes.

File: Samples/Caboodle.Samples/ViewModel/PhoneDialerViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+        async void OnOpenPhoneDialer()
+        {
+            try
+            {
+                PhoneDialer.Open(PhoneNumber);
+            }
+            catch (Exception)
+            {
+                await Application.Current.MainPage.DisplayAlert("Error", "Dialer is not supported.", "OK");
```
Comment (mattleibow) [inline Samples/Caboodle.Samples/ViewModel/PhoneDialerViewModel.cs]: I added a method `BaseViewModel.DisplayAlert` so you shouldn't need to get the current page.

File: DeviceTests/Caboodle.DeviceTests.Shared/PhoneDialer_Tests.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿using Microsoft.Caboodle;
+using Xunit;
+
+namespace Caboodle.DeviceTests
+{
+    public class PhoneDialer_Tests
+    {
+    }
+}
```
Comment (Redth) [inline DeviceTests/Caboodle.DeviceTests.Shared/PhoneDialer_Tests.cs]: Is this file necessary?

File: Samples/Caboodle.Samples/View/PhoneDialerPage.xaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
```
Comment (Redth) [inline Samples/Caboodle.Samples/View/PhoneDialerPage.xaml]: We're using `views:BasePage` elsewhere in the sample.  I think we should be consistent and switch to it here too...

File: DeviceTests/Caboodle.DeviceTests.Shared/PhoneDialer_Tests.cs (hunk 5 above)
Comment (jamesmontemagno) [reply DeviceTests/Caboodle.DeviceTests.Shared/PhoneDialer_Tests.cs]: I am going to add some Internal tests as soon as I can pull that up

File: Samples/Caboodle.Samples/ViewModel/HomeViewModel.cs (hunk 2 above)
Comment (jamesmontemagno) [reply Samples/Caboodle.Samples/ViewModel/HomeViewModel.cs]: bad merge, woops

File: docs/en/index.xml (hunk 3 above)
Comment (jamesmontemagno) [reply docs/en/index.xml]: will revert, not sure how that happened.

**Comment by mattleibow** [comment]
Looks good so far.

**Comment by jamesmontemagno** [comment]
Approving! <3
